crates/tracing/src/metrics.rs

```diff
@@ -8,88 +8,43 @@ pub fn configure_metrics(
     service_name: &str,
     otel_config: &wasmcloud_core::OtelConfig,
 ) -> anyhow::Result<()> {
-    use opentelemetry_otlp::{MetricsExporterBuilder, WithExportConfig};
+    use opentelemetry_otlp::{WithExportConfig, WithHttpConfig};
+    use opentelemetry_sdk::metrics::{PeriodicReader, SdkMeterProvider};
     use wasmcloud_core::OtelProtocol;
 
-    let builder: MetricsExporterBuilder = match otel_config.protocol {
+    let exporter = match otel_config.protocol {
         OtelProtocol::Http => {
             let client = crate::get_http_client(otel_config)
                 .context("failed to get an http client for otel metrics exporter")?;
-            opentelemetry_otlp::new_exporter()
-                .http()
-                .with_protocol(opentelemetry_otlp::Protocol::HttpBinary)
+            opentelemetry_otlp::MetricExporter::builder()
+                .with_http()
                 .with_http_client(client)
+                .with_protocol(opentelemetry_otlp::Protocol::HttpBinary)
                 .with_endpoint(otel_config.metrics_endpoint())
-                .into()
+                .build()
+                .context("failed to create OTEL http exporter")?
         }
         OtelProtocol::Grpc => {
             // TODO(joonas): Configure tonic::transport::ClientTlsConfig via .with_tls_config(...), passing in additional certificates.
-            opentelemetry_otlp::new_exporter()
-                .tonic()
+            opentelemetry_otlp::MetricExporter::builder()
+                .with_tonic()
                 .with_endpoint(otel_config.metrics_endpoint())
-                .into()
+                .build()
+                .context("failed to create OTEL tonic exporter")?
         }
     };
 
-    opentelemetry_otlp::new_pipeline()
-        .metrics(opentelemetry_sdk::runtime::Tokio)
-        .with_exporter(builder)
+    let reader = PeriodicReader::builder(exporter, opentelemetry_sdk::runtime::Tokio).build();
+
+    let meter_provider = SdkMeterProvider::builder()
         .with_resource(opentelemetry_sdk::Resource::new(vec![
             opentelemetry::KeyValue::new("service.name", service_name.to_string()),
         ]))
-        .with_aggregation_selector(ExponentialHistogramAggregationSelector::new())
-        .with_temporality_selector(
-            opentelemetry_sdk::metrics::reader::DefaultTemporalitySelector::new(),
-        )
-        .build()
-        .context("failed to create OTEL metrics provider")?;
-
-    Ok(())
-}
-
-/// Replaces the default `ExplicitBucketHistogram` aggregation for Histograms
-/// with `Base2ExponentialHistogram`. This makes it easier to capture latency
-/// at nanosecond accuracy.
-#[cfg(feature = "otel")]
-#[derive(Clone, Default, Debug)]
-struct ExponentialHistogramAggregationSelector {
-    pub(crate) _private: (),
-}
+        .with_reader(reader)
+        // .with_aggregation_selector(ExponentialHistogramAggregationSelector::new())
```

Comment:
Nit: Left over debug/refactor?

Reply:
Also, just double checking, we don't need any aggregation here, right?

Reply:
With [open-telemetry/opentelemetry-rust#2085](https://github.com/open-telemetry/opentelemetry-rust/pull/2085) this option was removed, so the aggregation will be based on https://opentelemetry.io/docs/specs/otel/metrics/sdk/#default-aggregation.

That said, I believe a future improvement that could be made here is to set custom values for the explicit histogram boundaries rather than relying on the defaults from the default aggregation.

It's not clear to me that there's an easy way to get the previous functionality back with the changes that were made upstream, there are features that were mentioned as being applicable for this ("views"), but even their future seems slightly uncertain, hence our best for the moment being getting setting explicit boundaries.